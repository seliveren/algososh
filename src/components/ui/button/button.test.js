import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, fireEvent} from '@testing-library/react';
import {Button} from './button';


describe('Test Button component', () => {
  it("button without text is rendered correctly", () => {
    const ButtonComponent = renderer.create(<Button/>).toJSON();
    expect(ButtonComponent).toMatchSnapshot();
  });

  it("button with text is rendered correctly", () => {
    const props = {
      text: "Кнопка"
    };
    render(<Button {...props} />);
    const ButtonComponent = renderer.create(<Button {...props}/>).toJSON();
    expect(screen.getByRole('button')).toHaveTextContent('Кнопка');
    expect(ButtonComponent).toMatchSnapshot();
  });

  it("disabled button is rendered correctly", () => {
    const props = {
      disabled: true
    };
    render(<Button {...props} />);
    const ButtonComponent = renderer.create(<Button {...props}/>).toJSON();
    expect(screen.getByRole('button')).toBeDisabled();
    expect(ButtonComponent).toMatchSnapshot();
  });

  it("loading button is rendered correctly", () => {
    const props = {
      isLoader: true
    };
    render(<Button {...props} />);
    const ButtonComponent = renderer.create(<Button {...props}/>).toJSON();
    expect(screen.getByAltText('Загрузка.')).not.toEqual(null);
    expect(ButtonComponent).toMatchSnapshot();
  });

  it("callback is executed when the button is pressed", () => {
    const callback = jest.fn();
    render(<Button text={"Click me!"} onClick={callback}/>);
    const button = screen.getByText("Click me!");
    fireEvent.click(button);
    expect(callback).toHaveBeenCalled();
  });

});


