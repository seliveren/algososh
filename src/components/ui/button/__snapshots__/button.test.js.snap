// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component button with text is rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;

exports[`Test Button component button without text is rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Test Button component disabled button is rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Test Button component loading button is rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
